src/main/java/org/jitsi/videobridge/BitrateController.java

```diff
@@ -0,0 +1,568 @@
+
+    /**
+     * Initializes a new {@link BitrateController} instance which is to
+     * belong to a particular {@link VideoChannel}.
+     *
+     * @param dest the {@link VideoChannel} that owns this instance.
+     */
+    public BitrateController(VideoChannel dest)
+    {
+        this.dest = dest;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTPTransformer()
+    {
+        return rtpTransformer;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTCPTransformer()
+    {
+        return rtcpTransformer;
+    }
+
+    /**
+     *
+     * @param data
+     * @param buf
+     * @param off
+     * @param len
+     * @param source
+     * @return
+     */
+    public boolean rtpTranslatorWillWrite(
+        boolean data, byte[] buf, int off, int len, RtpChannel source)
+    {
+        Integer ssrc;
+        if (data)
+        {
+
+            long ret = RawPacket.getSSRCAsLong(buf, off, len);
+            if (ret < 0)
+            {
+                return false;
+            }
+            ssrc = (int) ret;
+        }
+        else
+        {
+            // NOTE(gp) This will need to be adjusted when we enable
+            // reduced-size RTCP.
+            long ret = RTCPHeaderUtils.getSenderSSRC(buf, off, len);
+            if (ret < 0)
+            {
+                return false;
+            }
+
+            ssrc = (int) ret;
+        }
+
+        SimulcastController subStrl = subCtrls.get(ssrc);
+
+        return subStrl != null
+            && subStrl.rtpTranslatorWillWrite(data, buf, off, len);
+    }
+
+    /**
+     *
+     * @param conferenceEndpoints
+     */
+    public void update(List<Endpoint> conferenceEndpoints)
+    {
+        // Gather the conference allocation input.
+        if (conferenceEndpoints == null)
+        {
+            conferenceEndpoints
+                = dest.getConferenceSpeechActivity().getEndpoints();
+        }
+
+        long bweBps = Long.MAX_VALUE;
+
+        // Compute the bitrate allocation.
+        EndpointBitrateAllocation[] allocations
+            = allocate(bweBps, conferenceEndpoints);
+
+        List<String> newForwardedEndpoints = new ArrayList<>();
+
+        for (int i = 0; i < allocations.length; i++)
+        {
+            int ssrc = allocations[i].targetSSRC,
+                targetIdx = allocations[i].targetIdx;
+
+            SimulcastController ctrl = subCtrls.get(ssrc);
+            if (ctrl == null && allocations[i].track != null)
+            {
+                ctrl = new SimulcastController(allocations[i].track);
+
+                RTPEncodingImpl[] rtpEncodings
+                    = allocations[i].track.getRTPEncodings();
+
+                // Route all encodings to the specified bitrate controller.
+                for (int j = 0; j < rtpEncodings.length; j++)
+                {
+                    subCtrls.put((int) rtpEncodings[j].getPrimarySSRC(), ctrl);
```

Comment:
Casting a primitive long into a primitive int is dirty cheap (if not optimized out entirely by the compiler) compared to the unavoidable implicit cast of an int to Integer that takes place here..